api/team_members.go

```diff
@@ -64,26 +98,100 @@ func (svc *TeamMembersService) List() (res TeamMembersResponse, err error) {
+	}
+	err = svc.client.RequestEncoderDecoder("POST", apiV2TeamMembers, tm, &res)
+	return
+}
+
+// CreateOrg creates a single team member at the org level
+//TODO Move all ORG stuff into a different file
+func (svc *TeamMembersService) CreateOrg(tm TeamMemberOrg) (res TeamMemberOrgResponse, err error) {
+	if !svc.client.OrgAccess() {
+		return res, errors.New("client configured to manage account-level datasets, use Create()")
+	}
 	err = svc.client.RequestEncoderDecoder("POST", apiV2TeamMembers, tm, &res)
 	return
 }
 
-// Delete deletes a single team member with the corresponding guid
+// Delete deletes a single team member at the account level with the corresponding guid
 func (svc *TeamMembersService) Delete(guid string) error {
+	if svc.client.OrgAccess() {
+		return errors.New("client configured to manage org-level datasets, use DeleteOrg()")
+	}
 	if guid == "" {
 		return errors.New("please specify a guid")
 	}
+	return svc.client.RequestDecoder("DELETE", fmt.Sprintf(apiV2TeamMembersFromGUID, guid), nil, nil)
+}
 
+// DeleteOrg deletes a single team member at the org level with the corresponding guid
+func (svc *TeamMembersService) DeleteOrg(guid string) error {
+	if !svc.client.OrgAccess() {
+		return errors.New("client configured to manage account-level datasets, use Delete()")
+	}
+	if guid == "" {
+		return errors.New("please specify a guid")
+	}
 	return svc.client.RequestDecoder("DELETE", fmt.Sprintf(apiV2TeamMembersFromGUID, guid), nil, nil)
 }
 
-// Update updates a single team member with the corresponding guid
+// Update updates a single team member at the account-level with the corresponding guid
 func (svc *TeamMembersService) Update(tm TeamMember) (res TeamMemberResponse, err error) {
+	if svc.client.OrgAccess() {
+		return res, errors.New("client configured to manage org-level datasets, use UpdateOrg()")
+	}
 	if tm.UserGuid == "" {
 		err = errors.New("please specify a guid")
 		return
 	}
-	err = svc.client.RequestEncoderDecoder("PATCH", fmt.Sprintf(apiV2TeamMembersFromGUID, tm.UserGuid), tm, &res)
+	userGuid := tm.UserGuid
+	// Omit userGuid for patch requests
+	tm.UserGuid = ""
+	// Omit userName for patch requests as it cannot be modified
+	tm.UserName = ""
+	err = svc.client.RequestEncoderDecoder("PATCH", fmt.Sprintf(apiV2TeamMembersFromGUID, userGuid), tm, &res)
+	return
+}
+
+// UpdateOrg updates a single team member at the org-level with the corresponding username
+func (svc *TeamMembersService) UpdateOrg(tm TeamMemberOrg) (res TeamMemberOrgResponse, err error) {
+	if !svc.client.OrgAccess() {
+		return res, errors.New("client configured to manage account-level datasets, use Update()")
+	}
+	if tm.UserName == "" {
+		err = errors.New("please specify a username")
+		return
+	}
+	tms, errSearch := svc.SearchUsername(tm.UserName)
+	if errSearch != nil || len(tms.Data) == 0 {
+		err = errors.Wrap(err, "unable to find user with specified username")
+		return
+	}
+	return svc.UpdateOrgById(tm)
```

Comment:
Uhmmm I feel there is something missing here, you are just passing tm without modifying the struct with the userGuid found on the search. Am I wrong?